pkg/discovery/servicehandler.go

```diff
@@ -37,50 +38,41 @@ type ServiceHandler interface {
 }
 
 type serviceHandler struct {
-	stage               string
+	muleEnv             string
 	discoveryTags       []string
 	discoveryIgnoreTags []string
 	client              anypoint.Client
 	subscriptionManager subscription.SchemaHandler
+	cache               cache.Cache
 }
 
 func (s *serviceHandler) OnConfigChange(cfg *config.MulesoftConfig) {
 	s.discoveryTags = cleanTags(cfg.DiscoveryTags)
 	s.discoveryIgnoreTags = cleanTags(cfg.DiscoveryIgnoreTags)
-	s.stage = cfg.Environment
+	s.muleEnv = cfg.Environment
 }
 
 // ToServiceDetails gathers the ServiceDetail for a single Mulesoft Asset. Each Asset has multiple versions and
 // can resolve to multiple ServiceDetails.
 func (s *serviceHandler) ToServiceDetails(asset *anypoint.Asset) []*ServiceDetail {
 	serviceDetails := []*ServiceDetail{}
-	log := logrus.WithFields(logrus.Fields{
+	logger := logrus.WithFields(logrus.Fields{
 		"assetName": asset.AssetID,
 		"assetID":   asset.ID,
 	})
 	for _, api := range asset.APIs {
-		fields := logrus.Fields{
-			"apiID":           api.ID,
-			"apiAssetVersion": api.AssetVersion,
-		}
-
-		key := formatCacheKey(fmt.Sprint(api.ID), s.stage)
+		logger.
+			WithField("apiID", api.ID).
+			WithField("apiAssetVersion", api.AssetVersion)
 
-		// TODO Implement deletion of items from cache
-		err := cache.GetCache().Set(key, api)
-		if err != nil {
-			log.WithFields(fields).Error(err)
-		}
-
-		// TODO Handle purging of cache
-		err = cache.GetCache().SetWithSecondaryKey(key, strconv.FormatInt(api.ID, 10), api)
-		if err != nil {
-			log.WithFields(fields).Error(err)
+		if !shouldDiscoverAPI(api.EndpointURI, s.discoveryTags, s.discoveryIgnoreTags, api.Tags) {
```

Comment:
Moved this check to happen at the start of the loop to find out immediately if an API should be discovered or not.